tower-hedge/src/lib.rs

```diff
@@ -0,0 +1,199 @@
+extern crate futures;
```

Comment:
If you slap these attributes at the top, it catches a few common things we want to try to always provide:

```rust
#![deny(warnings)]
#![deny(missing_docs)]
#![deny(missing_debug_implementations)]
```

Reply:
Adding debug implementations for everything turned out to be a pain, so I didn't do it.  I've derived debug in a bunch more places though.

Reply:
FWIW, in some cases I just provide a debug impl that outputs the name of the struct and nothing else. I'd recommend having them on services, requests, and responses. Futures are whatever.